File: challenges/20/solution.ts
function isValid(s: string): boolean {
  let myStack: string[] = [];
  for (let c of s) {
    if (c === "{" || c === "[" || c === "(") {
      myStack.push(c);
    } else {
      if (myStack.length === 0) return false;
      const popped = myStack.pop();
      if (popped === "[" && c !== "]") return false;
      if (popped === "{" && c !== "}") return false;
      if (popped === "(" && c !== ")") return false;
    }
  }
  return !myStack.length;
}

File: challenges/20/solution.py

def isValid(self, s: str) -> bool:
    stack = []

    # push open brackets
    for c in s:
        if c in ['(', '[', '{']:
            stack.append(c)
        else:
            
            # avoid lonely closed brackets 
            if stack:
                open = stack.pop(-1)
            else:
                return False
            # check close brackets
            if c is ')' and open is '(' or \
                c is ']' and open is '[' or \
                c is '}' and open is '{':
                continue
            else:
                return False
    
    # avoid lonely open brackets
    if stack:
        return False
    return True
                
